adoptingerlang.org

```diff
@@ -1334,6 +1334,370 @@ Repeat this exercise from time to time, and eventually do some [[https://princip
+- Builds are meant to be repeatable. Dependencies are locked, and only upgraded when specifically asked to. The =rebar3.lock= file will overtake whatever versions are declared in =rebar.config= until =rebar3 upgrade= is called to change them.
+- Dependencies are always built with the =prod= profile applied, and once built will not be recompiled unless they are upgraded, even though the tool will always make sure they are there, match the lock file, and are compiled. Here "compiled" means that there is a =.app= file and for each module listed in the app file a corresponding beam file is found, but it does _not_ check if there are changes to the source file for that beam module.
+- Rebar3 assumes that you'll sometimes need to tweak the configuration of libraries you do not control and supports [[https://www.rebar3.org/docs/configuration#section-overrides][overrides]] for this purpose
+- The tool assumes you're developing with a source control mechanism such as =git= or =hg= (mercurial), which means that switching branches may switch dependency versions in the lock file. Since rebar3 verifies dependencies before each build, it will automatically re-fetch libraries to get the locked version for the current branch if there is a change when switching branches.
+- To ease contribution and publishing, rebar3 does not natively support ways to use relative paths to declare libraries, since this could make builds brittle, non-repeatable, and non-portable when publishing code
+- Knowing relative paths are very common when making changes within the dependencies of a project, [[https://www.rebar3.org/docs/dependencies#section-checkout-dependencies][=_checkouts=]] allow for an automatic way to temporary override of a dependency with a local copy. For other use cases [[https://www.rebar3.org/docs/using-available-plugins#section-vendoring-dependencies][plugins]] allow the creation of [[https://www.rebar3.org/docs/custom-dep-resources][custom resource types]].
+- Rebar3 is not an installer of end-user applications and does not support anything related to that; it aims to generate build artifacts that you can then install through the correct dedicated channels. Rebar3 does not expect or need to ever be on a production device or server.
+- Rebar3 is not a sandbox. While it will make sure all dependencies you download match the right signatures and provides repeatable builds, it cannot guarantee that script files invoked during builds or [[https://stackoverflow.com/questions/2416192/is-there-a-good-complete-tutorial-on-erlang-parse-transforms-available][parse transforms]] run during compilation are ever going to be safe, and has no intention of taking on that responsibility.
```

Comment:
May want to mention there are caveats to "repeatable builds" here, or maybe just say "attempts to provide repeatable builds" since this isn't the place to detail issues like plugins not being locked and shit like that.